### PR TITLE
DH Key and Params Export cleanups and Apache httpd fixes

### PR DESCRIPTION
**Pull request by dgarske**
* Refactor of DH key and param exports code (moved into asn.c) enabled with `WOLFSSL_DH_EXTRA`.
* Cleanup `WOLFSSL_DH_EXTRA` macro logic and do not allow with FIPS v1 or v2.
* Fixes for httpd (if `SSL_CONF_FLAG_FILE` is defined it is used to indicate support for `SSL_CONF_CTX_set_flags` and `SSL_CONF_cmd_value_type`).
* Add Curve448 and ED448 key type to `enum wc_PkType`.
* Expand `dh_ffdhe_test` to include 4096 bit.

### CODE REVIEW

**Review by SparkiDev — CHANGES_REQUESTED** [review]
Not seeing a test that calls the new encoding functions.

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -16652,67 +16658,120 @@ WOLFSSL_TEST_SUBROUTINE int dh_test(void)
...
+    #endif
+        idx = 0;
+        XMEMSET(tmp2, 0, DH_TEST_TMP_SIZE);
+
+        /* Import DH Private key as DER */
+        ret = wc_DhKeyDecode(tmp, &idx, key, bytes);
+        if (ret == 0) {
+            /* Export as DER */
+            idx = DH_TEST_TMP_SIZE;
+            ret = wc_DhPrivKeyToDer(key, tmp2, &idx);
```
Comment (dgarske) [inline wolfcrypt/test/test.c:16688]: @SparkiDev , The `wc_DhPrivKeyToDer` function calls both `wc_DhKeyToDer` and `wc_DhParamsToDer`.

File: wolfcrypt/test/test.c [hunk 2]
```diff
@@ -16652,67 +16658,120 @@ WOLFSSL_TEST_SUBROUTINE int dh_test(void)
...
+
+        idx = 0;
+        XMEMSET(tmp2, 0, DH_TEST_TMP_SIZE);
+
+        /* Import DH Public key as DER */
+        ret = wc_DhKeyDecode(tmp, &idx, key, bytes);
+        if (ret == 0) {
+            /* Export as DER */
+            idx = DH_TEST_TMP_SIZE;
+            ret = wc_DhPubKeyToDer(key, tmp2, &idx);
```
Comment (dgarske) [inline wolfcrypt/test/test.c:16712]: @SparkiDev , The `wc_DhPubKeyToDer` function calls both `wc_DhKeyToDer` and `wc_DhParamsToDer`.

**Review by SparkiDev — APPROVED** [review]

File: wolfcrypt/src/asn.c [hunk 3]
```diff
@@ -4666,6 +4666,156 @@ int wc_DhKeyDecode(const byte* input, word32* inOutIdx, DhKey* key, word32 inSz)
...
+    /* determine size */
+    if (exportPriv) {
+        /* octect string: priv */
+        privSz = SetASNIntMP(&key->priv, -1, NULL);
+        idx = 1 + SetLength(privSz, NULL) + privSz; /* +1 for ASN_OCTET_STRING */
+    }
+    else {
+        /* bit string: public */
+        pubSz = SetASNIntMP(&key->pub, -1, NULL);
+        idx = 1 + SetLength(pubSz, NULL) + pubSz; /* +1 for ASN_BIT_STRING */
```
Comment (SparkiDev) [inline wolfcrypt/src/asn.c:4690]: BIT_STRING has an extra byte to indicate the number of unused bits in last octet.
Use SetBitString(pubSz, 0, NULL) to get length if possible.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Review by SparkiDev — APPROVED** [review]

**Comment by kaleb-himes** [comment]
retest this please

**Pull request merged**